plugins/event_notification/providers/push/services/PushNotificationTemplateService.php

```diff
@@ -7,96 +7,109 @@
+		
+		$postProcessor = new registerNotificationPostProcessor();
+		if(kApiCache::getEnableResponsePostProcessor())
+			kApiCache::setResponsePostProcessor($postProcessor);
 
-	    // build the url to return
-	    $protocol = infraRequestUtils::getProtocol();
-	    $host = kConf::get("push_server_host");
-	    $secret = kConf::get("push_server_secret");
-	    $ip = kCurrentContext::$user_ip;
-	    $token = base64_encode($partnerId . ":" . $this->encode($secret . ":" . $ip . ":" . $hash . ":" . uniqid() ) );
-	    $result->url = $protocol . "://" . $host ."/?p=" . $partnerId ."&x=" . urlencode($token);
-	    
-	    return $result;
+		/* @var $kPushNotificationParams kPushNotificationParams */
+		$missingParams = array();
+		$userContentParams = array();
+		$userQueueKeyParams = array();
+		$userParamsArrayKeys = array();
+		$userParamsArray = $pushNotificationParams->toObject()->getUserParams();
+		
+		// get template configured params
+		$contentParams = $dbEventNotificationTemplate->getContentParametersKeyValueArray();
+		$queueKeyParams = $dbEventNotificationTemplate->getQueueKeyParametersKeyValueArray();
+		$templateParams = array_merge($contentParams, $queueKeyParams);
+		
+		foreach ($userParamsArray as $userParam)
+		{
+			$userParamKey = $userParam->getKey();
+			array_push($userParamsArrayKeys, $userParamKey);
+			
+			if(isset($contentParams[$userParamKey]))
+			{
+				$userContentParams[] = $userParam;
+				continue;
+			}
+			
+			if(isset($queueKeyParams[$userParamKey]))
+			{
+				$valueToken = $queueKeyParams[$userParamKey]->getQueueKeyToken();
+				if($valueToken && kApiCache::getEnableResponsePostProcessor())
+				{
+					$userParam->setValue(new kStringValue($valueToken));
+					$postProcessor->addToken($userParamKey, $valueToken);
+				}
+				$userQueueKeyParams[] = $userParam;
+			}
+		}
+		
+		foreach ($templateParams as $templateParamKey => $templateParamValue)
+		{
+			if (!in_array($templateParamKey, $userParamsArrayKeys))
+				array_push($missingParams, $templateParamKey);
+		}
+		
+		if ($missingParams != null)
+			throw new KalturaAPIException(KalturaErrors::MISSING_MANDATORY_PARAMETER, implode(",", $missingParams));
+		
+		$queueName = $dbEventNotificationTemplate->getQueueName($userContentParams, $partnerId, null);
+		$queueKey = $dbEventNotificationTemplate->getQueueKey(array_merge($userContentParams, $userQueueKeyParams), $partnerId, null, kApiCache::getEnableResponsePostProcessor());
+		
+		return $postProcessor->buildResponse($partnerId, $queueName, $queueKey);
+	}
+	
+	/**
+	 * Clear queue messages 
+	 *
+	 * @action sendCommand
+	 * @actionAlias eventNotification_eventNotificationTemplate.sendComman
+	 * @param string $queueName QueueNAme to clear messages for
+	 * @param KalturaPushNotificationCommandType $command Command to be sent to push server
+	 * @return bool Allwyas true
+	 */
+	function sendCommandAction($queueName, KalturaPushNotificationCommandType $command)
```

Comment:
You should use id or system-name of the template, using the plain queue name by-passes any security and partners default criterias.